common/fitparser.cpp

```diff
@@ -0,0 +1,1090 @@
+#include "fitparser.h"
+
+#ifdef U_ENABLE_FIT_PARSING_SUPPORT
+
+#include "intel_fit.h"
+#include "ffs.h"
+#include "parsingdata.h"
+#include "types.h"
+#include "utility.h"
+#include "digest/sha2.h"
+
+#include <sstream>
+#include "kaitai/kaitaistream.h"
+#include "generated/intel_acbp_v1.h"
+#include "generated/intel_acbp_v2.h"
+#include "generated/intel_keym_v1.h"
+#include "generated/intel_keym_v2.h"
+#include "generated/intel_acm.h"
+
+USTATUS FitParser::parseFit(const UModelIndex & index)
+{
+    // Reset parser state
+    fitTable.clear();
+    securityInfo = "";
+    bgAcmFound = false;
+    bgKeyManifestFound = false;
+    bgBootPolicyFound = false;
+    bgKmHash = UByteArray();
+    bgBpHashSha256 = UByteArray();
+    bgBpHashSha384 = UByteArray();
+    
+    // Check sanity
+    if (!index.isValid()) {
+        return U_INVALID_PARAMETER;
+    }
+    
+    // Search for FIT
+    UModelIndex fitIndex;
+    UINT32 fitOffset;
+    findFitRecursive(index, fitIndex, fitOffset);
+    
+    // FIT not found
+    if (!fitIndex.isValid()) {
+        // Nothing to parse further
+        return U_SUCCESS;
+    }
+    // Explicitly set the item containing FIT as fixed
+    model->setFixed(fitIndex, true);
+    
+    // Special case of FIT header
+    UByteArray fitBody = model->body(fitIndex);
+    const INTEL_FIT_ENTRY* fitHeader = (const INTEL_FIT_ENTRY*)(fitBody.constData() + fitOffset);
```

Comment:
Where does this check for oob access?

Reply:
Added to findFitRecursive, thanks for catching it.